h/groups/views.py

```diff
@@ -101,26 +104,28 @@ def _read_group(request, group):
     """
     url = request.route_url('group_read', pubid=group.pubid, slug=group.slug)
 
-    result = search.search(request,
-                           private=False,
-                           params={"group": group.pubid, "limit": 1000})
-    annotations = [presenters.AnnotationHTMLPresenter(
-                       storage.annotation_from_dict(a))
-                   for a in result['rows']]
-
-    # Group the annotations by URI.
-    # Create a dict mapping the (normalized) URIs of the annotated documents
-    # to the most recent annotation of each document.
-    annotations_by_uri = collections.OrderedDict()
-    for annotation in annotations:
-        normalized_uri = uri.normalize(annotation.uri)
-        if normalized_uri not in annotations_by_uri:
-            annotations_by_uri[normalized_uri] = annotation
-            if len(annotations_by_uri) >= 25:
-                break
-
-    document_links = [annotation.document_link
-                      for annotation in annotations_by_uri.values()]
+    # Find documents that have at least one annotation that:
+    # a. is shared, and
+    # b. is in this group.
+    documents = (
```

Comment:
Ok the above query was incorrect, I've changed it to something completely different. Unfortunately I don't think there's any correct and efficient way to get a group's 25 most recently annotated documents currently, but the way that I've changed it to is more or less as good ad the previous Elasticsearch solution (it might be slower than Es on groups with lots of annotations).